src/sentry/search/events/datasets/metrics.py

```diff
@@ -240,22 +337,71 @@ def _resolve_failure_count(
                             "equals",
                             [
                                 Column("metric_id"),
-                                self.resolve_metric("transaction.duration"),
+                                args["metric_id"],
```

Comment:
Should this also be hardcoded since it's always going to be user? but it's nice to maintain the same interface as the discover snql function... Maybe as a follow up, should we have an allowlist of metrics for this so that we error early if people try to access metrics that won't be supported for this function? I'm assuming right now if I did something like `count_miserable(transaction.duration)` it would run it like a valid query and return 0?

Reply:
Hmmm good point I'll backlog something to allowlist the columns for each function